annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +306,287 @@ def test_validate_user_actions_invalid_states(
+    request = TaskFilter()
+    tenant = "test_tenant"
+    token = "test_token"
+    with patch(
+        "annotation.tasks.services.map_request_to_filter"
+    ) as mock_map_request_to_filter, patch(
+        "annotation.tasks.services.remove_additional_filters"
+    ) as mock_remove_additional_filters, patch(
+        "annotation.tasks.services.get_files_by_request"
+    ) as mock_get_files_by_request, patch(
+        "annotation.tasks.services.get_jobs_by_name"
+    ) as mock_get_jobs_by_name, patch(
+        "annotation.tasks.services.form_query"
+    ) as mock_form_query, patch(
+        "annotation.tasks.services.paginate"
+    ) as mock_paginate:
+        mock_query = MagicMock()
+        mock_session.query.return_value = mock_query
+
+        mock_map_request_to_filter.return_value = {
+            "filters": [],
+            "sorting": [],
+        }
+        mock_remove_additional_filters.return_value = additional_filters
+        mock_get_files_by_request.return_value = files_by_name
+        mock_get_jobs_by_name.return_value = jobs_by_name
+
+        mock_query.filter.return_value = mock_query
+        mock_form_query.return_value = (MagicMock(), MagicMock())
+        mock_paginate.return_value = ([MagicMock()], MagicMock())
+
+        result = filter_tasks_db(mock_session, request, tenant, token)
+
+        assert len(result[0][0]) == len(expected_result[0])
+        assert result[1] == expected_result[1]
+        assert result[2] == expected_result[2]
+
+
+@pytest.mark.parametrize(
+    ("additional_filters"),
+    (({"file_name": ["file1"]}), ({"job_name": ["job1"]})),
+)
+def test_filter_tasks_db_no_files_or_jobs(
+    mock_session: Mock,
+    additional_filters: Dict[str, List[str]],
+):
+    files_by_name = {}
+    jobs_by_name = {}
+    expected_result = ([], {}, {})
+    request = TaskFilter()
+    tenant = "test_tenant"
+    token = "test_token"
+    with patch(
+        "annotation.tasks.services.map_request_to_filter"
+    ) as mock_map_request_to_filter, patch(
+        "annotation.tasks.services.remove_additional_filters"
+    ) as mock_remove_additional_filters, patch(
+        "annotation.tasks.services.get_files_by_request"
+    ) as mock_get_files_by_request, patch(
+        "annotation.tasks.services.get_jobs_by_name"
+    ) as mock_get_jobs_by_name, patch(
+        "annotation.tasks.services.form_query"
+    ) as mock_form_query, patch(
+        "annotation.tasks.services.paginate"
+    ) as mock_paginate:
+        mock_query = MagicMock()
+        mock_session.query.return_value = mock_query
+
+        mock_map_request_to_filter.return_value = {
+            "filters": [],
+            "sorting": [],
+        }
+        mock_remove_additional_filters.return_value = additional_filters
+        mock_get_files_by_request.return_value = files_by_name
+        mock_get_jobs_by_name.return_value = jobs_by_name
+
+        mock_query.filter.return_value = mock_query
```

Comment:
please check out patch different use cases. It was fine before to get familiar with mock and patches, but now lets make it cleaner - you can do everything in the patch.

Reply:
right now you basically split mocking into two parts - patch and mock attributes, which is unnecessary in your case